app/questionnaire/questionnaire_manager.py

```diff
@@ -11,11 +16,24 @@ def __init__(self, schema, response_store, validator, validation_store, routing_
         self._routing_engine = routing_engine
         self._navigator = navigator
         self._navigation_history = navigation_history
+        self.renderer = Renderer(self._schema, self._response_store, self._validation_store, self._navigator)
 
     def process_incoming_responses(self, post_data):
         # process incoming post data
-        user_responses = self._process_incoming_post_data(post_data)
-
+        user_action, user_responses = self._process_incoming_post_data(post_data)
+
+        # Are we submitting downstram?
```

Comment:
submit_answer should be constant. I would like to see the submission refactored into its own method